### PR TITLE
Replace exec --cache with --vendor-cache

### PR DESCRIPTION
**Pull request by jquick**
Fixes https://github.com/chef/inspec/issues/2339

This change deprecates the current `--cache` exec flag in favor of the `--vendor-cache` flag. Deprecation warnings have been added and code updated.

Signed-off-by: Jared Quick <jquick@chef.io>

### CODE REVIEW

File: lib/inspec/base_cli.rb [hunk 1]
```diff
@@ -64,6 +64,8 @@ def self.exec_options
       option :attrs, type: :array,
         desc: 'Load attributes file (experimental)'
       option :cache, type: :string,
+        desc: '[DEPRECATION] Please use --vendor-cache - this will be removed in InSpec 2.0'
```
Comment (adamleff) [inline lib/inspec/base_cli.rb]: Can we change this to `[DEPRECATED]`? It's cleared to the users that it's deprecated, vs. us declaring that it's a deprecation. A nitpick, I know, but grammar matters :)

File: lib/inspec/cli.rb [hunk 2]
```diff
@@ -155,6 +155,13 @@ def exec(*targets)
     o = opts(:exec).dup
     configure_logger(o)
 
+    # check for deprecated --cache
+    # TODO: REMOVE for inspec 2.0
+    if o.key?('cache')
+      o[:vendor_cache] = o[:cache]
+      o[:logger].warn '[DEPRECATION] The use of `--cache` is being deprecated in InSpec 2.0. Please use `--vendor-cache` instead.'
```
Comment (adamleff) [inline lib/inspec/cli.rb]: Same here - `DEPRECATED` instead of `DEPRECATION`

**Review by adamleff — CHANGES_REQUESTED** [review]
@jquick this looks great. Thanks for taking the time to cleanup the keys throughout the code as well.

Just a couple of grammar nits, but otherwise this is great.

**Review by adamleff — APPROVED** [review]

**Review by arlimus — APPROVED** [review]
Great to be so clear on this now! Kudos 😁

**Pull request merged**